src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1081,6 +1082,14 @@ protected virtual void RegisterNode(NodeModel node)
             }
         }
 
+        private void Node_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if(e.PropertyName == "IsSetAsInput" || e.PropertyName == "IsSetAsOutput")
+            {
+                this.HasUnsavedChanges = true;
```

Comment:
@aparajit-pratap Is this more of what you meant by setting the DYN file as modified when a node is set as an input or an output node?

Reply:
Yes.

Reply:
@alfarok Can you add a gif? I think we are clear that after marking node as Input/output:
1. Graph does not re-run
2. Graph flagged as touched, so when closing Dynamo, it asks user to save

Reply:
@QilongTang added a gif, should be working as expected now without re-executing the workspace

Reply:
@alfarok I'm still confused by this. Other nodes must be doing this same thing somewhere else, why is this new handler necessary?

Reply:
@alfarok  - a quick look and I can't find what I thought existed ;) - but I am still curious how setting Name essentially works and setting these properties requires a new handler?

If you decide to keep this handler please look into it being unsubscribed when the workspace is disposed.

Reply:
@mjkkirschner made a few changes, let me know what you think.